cmd/kube-apiserver/app/server.go

```diff
@@ -339,7 +341,13 @@ func (s *APIServer) Run(_ []string) error {
 		glog.Fatalf("Invalid server address: %v", err)
 	}
 
-	etcdStorage, err := newEtcd(s.EtcdConfigFile, s.EtcdServerList, latest.GroupOrDie("").InterfacesFor, latest.GroupOrDie("").Version, s.StorageVersion, s.EtcdPathPrefix)
+	g, err := latest.Group("")
+	if err != nil {
+		return err
+	}
+	storageVersions := make(map[string]string)
```

Comment:
> Do we really need to call newEtcd for every group?

I think so. newEtcd take InterfacesFor as an argument, which is different for every group.